contrib/pg_tde/documentation/docs/replication.md

```diff
@@ -1,4 +1,257 @@
-# Replication
+<!--**Extension Name:** The extension is `pg_tde`.
+*   **Table Access Method:** The encrypted table access method is `heap_tde`.
+*   **Key Provider GUC:** `pg_tde_key_provider` (e.g., `'file'`, `'vault'`).
+*   **File Provider Path GUC:** `pg_tde_file_keys_path` (points to a *directory* where key files are stored, not a single key file).
+*   **Principal  Key Name GUC:** `pg_tde_principal_key_name` (this is the logical name of the key, which then maps to a file in `pg_tde_file_keys_path` or an entry in Vault).
+*   **Key Management Functions:**
+    *   `pg_tde_principal_key_add(key_name TEXT, key_value TEXT, key_provider TEXT)`: Adds a key definition. `key_value` can be `NULL` if fetching from an external provider.
+    *   `pg_tde_set_principal_key(key_name TEXT, provider_name TEXT)`: Sets the active principal key for the cluster.
+*   **Vault GUCs:** `pg_tde_vault_url`, `pg_tde_vault_token`, `pg_tde_vault_mount_path`, `pg_tde_vault_ca_path`. -->
 
-<!-- To add information on Replication affecting pg_tde for customers and how it interacts with it
-Also, don't forget to add it to TOC>
\ No newline at end of file
+# Streaming Replication with pg_tde (heap_tde)
+
+This section outlines how to set up PostgreSQL streaming replication when Percona's `pg_tde` extension (specifically the `heap_tde` access method) is enabled on the primary server.
+
+!!! note
+    Physical streaming replication copies data blocks directly from the primary to the standby. If these blocks are encrypted on the primary, they arrive **encrypted** on the standby. Therefore, the standby **must** have access to the **exact same encryption keys** as the primary to decrypt and read this data.
+
+## Concepts
+
+* **`pg_tde`:** Percona's extension for Transparent Data Encryption in PostgreSQL.
+* **`heap_tde`:** The table access method provided by `pg_tde` that encrypts entire tables using a principal key.
+* **principal Key:** The central key used by `pg_tde` to encrypt/decrypt data. Identified by a `pg_tde_principal_key_name`.
+* **Key Provider:** The mechanism `pg_tde` uses to fetch the principal key (e.g., local files, HashiCorp Vault). Configured via `pg_tde_key_provider`.
```

Comment:
There's no method named `pg_tde_key_provider`, only multiple methods with somewhat similar names.

We also support KMIP